### ISSUE TITLE
Introduce LeaugesRepository

### ISSUE THREAD

**Issue by ilkeraslan** [ilkeraslan/dota2compose#21]
## Scope

We would like to show the available dota2 leagues information on a dedicates screen, thus we need a repository to fetch the data.

## User Stories

As a user, I can:
- [ ] Get a list of `League`s.
- [ ] Get a given `League`.
- [ ] Get the matches of a `League`.
- [ ] Get the teams of a `League`.

## Endpoints

The repository will return null for each method for now.

**Comment by indyfromoz** [comment]
Hi @ilkeraslan I am very keen to work on this and #22 . 
Can you please assign them both to me?

**Comment by ilkeraslan** [comment]
> Hi @ilkeraslan I am very keen to work on this and #22 . Can you please assign them both to me?

Hey @indyfromoz, thank you for the interest. For now I would say that you can work only on 1 issue in order to give the others the chance to contribute as well. If nobody show interest for #22, then it is yours 😆

**Comment by indyfromoz** [comment]
Hi @ilkeraslan So, I started working on this issue and find the dependencies used in the project are at least a year or more old. Do you have plans to update the dependencies soon? I prefer a separate issue for the upgrade.

**Comment by ilkeraslan** [comment]
> Hi @ilkeraslan So, I started working on this issue and find the dependencies used in the project are at least a year or more old. Do you have plans to update the dependencies soon? I prefer a separate issue for the upgrade.

Yeah I have to update the dependencies at some point, maybe we can create a separate issue for that! Thanks for the suggestion.

**Comment by indyfromoz** [comment]
@ilkeraslan Are you able to create an issue to update the dependencies? I am currently blocked on working with the code as a lot of dependencies need to be updated and the code requires fixes.

**Comment by indyfromoz** [comment]
> > Hi @ilkeraslan So, I started working on this issue and find the dependencies used in the project are at least a year or more old. Do you have plans to update the dependencies soon? I prefer a separate issue for the upgrade.
> 
> Yeah I have to update the dependencies at some point, maybe we can create a separate issue for that! Thanks for the suggestion.

Thanks for updating the dependencies. I will get on with the updates in the fork and submit a PR soon.
